src/ext/peek-legit.lisp

```diff
@@ -0,0 +1,369 @@
+    (setf *source-window* source-window)
+
+    (setf (current-window) peek-window)
+    (peek-legit-mode t)
+
+    (start-move-point (buffer-point (collector-buffer collector)))
+    (show-matched-line)))
+
+(defun make-peek-legit-buffer ()
+  (let ((buffer (make-buffer "*peek-legit*"
+                             :temporary t
+                             :enable-undo-p t
+                             :directory (uiop:getcwd))))
+    (setf (variable-value 'line-wrap :buffer buffer) nil)
+    buffer))
+
+(defun call-with-collecting-sources (function &key read-only)
+  (let* ((*collector* (make-instance 'collector :buffer (make-peek-legit-buffer)))
+         (point (buffer-point (collector-buffer *collector*))))
+    (declare (ignorable point))
+    (funcall function *collector*)
+    (when read-only
+      (setf (buffer-read-only-p (collector-buffer *collector*)) t))
+      (display *collector*)))
+
+(defmacro with-collecting-sources ((collector &key (read-only t)) &body body)
+  `(call-with-collecting-sources (lambda (,collector)
+                                   (declare (ignorable ,collector))
+                                   ,@body)
+                                 :read-only ,read-only))
+
+(defun call-with-appending-source (insert-function
+                                   move-function
+                                   visit-file-function
+                                   stage-function
+                                   unstage-function)
+  (let ((point (buffer-point (collector-buffer *collector*))))
+    (with-point ((start point))
+      (funcall insert-function point)
+      (unless (start-line-p point)
+        (insert-string point (string #\newline) :read-only t))
+      (set-move-function start point move-function)
+      (set-visit-file-function start point visit-file-function)
+      (set-stage-function start point stage-function)
+      (set-unstage-function start point unstage-function))
+    (incf (collector-count *collector*))))
+
+(defmacro with-appending-source ((point &key move-function
+                                             visit-file-function
+                                             stage-function
+                                             unstage-function) &body body)
+  `(call-with-appending-source (lambda (,point) ,@body)
+                               ,move-function
+                               ,visit-file-function
+                               ,stage-function
+                               ,unstage-function))
+
+(defun collector-insert (s &optional (newline t) header)
+  (let ((point (buffer-point (collector-buffer *collector*))))
+    (with-point ((start point))
+      (character-offset start 1)
+      (insert-string point s :read-only t)
+      (when header
+        (put-text-property start point 'header-marker t))
+      (when newline
+        (insert-string point (string #\newline) :read-only t)))))
+
+;;;
+(define-attribute match-line-attribute
+  (t :background "#444444"))
```

Comment:
This might be difficult to see depending on the theme.
How about using per base02?